swayfmt/tests/mod.rs

```diff
@@ -504,24 +503,23 @@ pub struct Foo { // Here is a comment
              //     \|__|    \|_______|\|_______|\|_______|        \|_______|\|__|\|__|\|_______|\_________\
              //                                                                                  \|_________|
 }
-// This is a comment
 "#,
         r#"contract;
 // This is a comment, for this one to be placed correctly we need to have Module visitor implemented
 pub struct Foo { // Here is a comment
+
```

Comment:
A heuristic I stuck to while rewriting the logic was to allow at most 1 newline if there are multiple newlines between comments. Should we actually squash multiline comments that are newline separated?

Reply:
> Should we actually squash multiline comments that are newline separated?

I think what you currently have (leaving just one empty line if one or more empty lines previously existed) is the right approach :+1:

Reply:
I think we had some kind of threshold before that the user could set to change the number of newlines between comments and items if they wanted to. But looking back on it, I don't really see anyone actually wanting to use that feature so this is good as it is.